Add docs for index.html and SPA frameworks client side routers (#786)
CleverCloud/doc.clever-cloud.com: deploy/application/static/static.md

{{< readfile "/content/partials/link-addon.md" >}}

{{< readfile "/content/partials/more-config.md" >}}

## Serving index.html for SPA (Single Page Application) routers

When you work with an SPA framework like React, Vue.js, Angular..., you're using client side routing.
This means when you click on a link going to `/the-page`, your browser doesn't make an HTTP request for `/the-page`.
Instead, the client side router highjacks the clicks on links, changes the DOM to display the page and ask the browser to change the URL in the address bar to `/the-page`.

What happens if you try to refresh the page?
If you do this, the browser will try to make an HTTP request for `/the-page`.
In most situations, SPA only have one HTML document at the root called `index.html`.
This is why, you'll probably get a 404 error.

To fix this, most people using SPA frameworks configure their HTTP server to serve the `index.html` for all unkown requests.
By this we mean for all requests that don't have a matching file on disk to serve.

To do this with our static applications, you need a `.htaccess` file like this at the root of your project:

```
RewriteEngine On

> # If an existing asset or directory is requested go to it as it is

Reviewer: ```suggestion
# If an existing asset or directory is requested, serve it
```
